### PR TITLE
Updating versioning guidelines

### PR DESCRIPTION
**Pull request by alefranz**
# Description of the Change

Updating versioning guidelines as part of [ADR-2024-01-29 API Versioning & Documentation Versions](https://newdaycards.atlassian.net/wiki/spaces/DA/pages/3566436359/ADR-2024-01-29+API+Versioning+Documentation+Versions?focusedCommentId=3603693709)

### CODE REVIEW

File: standards/Versioning.md [hunk 1]
```diff
@@ -101,28 +134,30 @@ Teams may define backwards compatibility as their business product line needs re
 Our guide on what broadly constitutes a breaking change is as follows:
 
 - Removing or renaming APIs or API parameters or headers
-- Removing or renaming responses fields or headers
-- Changes in behaviour for an existing API; including default parameter / argument behaviour
+- Removing or renaming response fields or headers
+- Changes in behavior for an existing API; including default parameter and argument behavior
```
Comment (dturveyza) [inline standards/Versioning.md:138]: American spelling of behaviour, is this intentional?

File: standards/Versioning.md [hunk 2]
```diff
@@ -101,28 +134,30 @@ Teams may define backwards compatibility as their business product line needs re
...
 - Changes in casing of property names, at any point in the object graph (:warning: beware if assigning objects directly based on responses from other services)
 - Changes in casing of headers
```
Comment (dturveyza) [inline standards/Versioning.md:140]: I thought the http header spec states they aren't case sensitive?

File: standards/Versioning.md [hunk 3]
```diff
@@ -101,28 +134,30 @@ Teams may define backwards compatibility as their business product line needs re
...
 
-**Objection**: It makes it hard to route traffic
-**Response**: You have several options for header based versioning.
+  **Response**: You have several options for header-based versioning.
 
-1. In application versioning - you can use the header inside of your application to route your requests in most common API frameworks. You can achieve this in ASP.NET Core.
+  1. In-application versioning - you can use the header inside of your application to route your requests in most common API frameworks. You can achieve this in ASP.NET Core.
 
-2. Outside of application versioning using APIM - you can use APIM dynamic routing to route to multiple running applications or functions based on a version header, if you prefer.
+  2. Outside of application versioning using APIM - you can use APIM dynamic routing to route to multiple running applications or functions based on a version header.
```
[the remark this answers is not in the document]
Comment (alefranz) [reply standards/Versioning.md:155]: you are right, we should be more generic

File: standards/Versioning.md [hunk 4]
```diff
@@ -1,15 +1,17 @@
 # Versioning
 
-All our API’s should follow a versioning protocol.
+All our APIs should follow a versioning protocol.
 
-Only major breaking changes should result in a new version, however, we expect developers to minimise the need for versioning by practicing good “addition” to API’s rather than creating a new version for every change. If items need to be made obsolete later it should be only removed when there is a major version change.
+Only major breaking changes should result in a new version. However, we expect developers to minimize the need for versioning by practicing good "addition" to APIs, rather than creating a new version for every change. If items need to be made obsolete later, they should only be removed when there is a major version change.
```
[the remark this answers is not in the document]
Comment (alefranz) [reply standards/Versioning.md:5]: I'll fix my spell checker :)

File: standards/Versioning.md (hunk 2 above)
Comment (alefranz) [reply standards/Versioning.md:140]: headers are not case sensitive per spec but they de facto are, many things broke historical due to header casing changes.
So we defined a convention in the naming standard

**Review by deniz-newday — APPROVED** [review]

**Pull request merged**